### PR TITLE
[CAP] OAuth2ServiceConfiguration simplification for implementations

### LINKED ISSUE #658: [CAP] OAuth2ServiceConfiguration improvements
.... in order to simplify creation via `OAuth2ServiceConfigurationBuilder` or creation from scratch

### PR DESCRIPTION
**Pull request by nenaraab**
closes #658

### CODE REVIEW

File: spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/XsuaaServiceConfigurationCustom.java [hunk 1]
```diff
@@ -72,4 +63,17 @@ public CredentialType getCredentialType() {
...
 	}
 
+	@Override
+	public String getProperty(String name) {
+		return null;
+	}
+
+	@Override
+	public ClientIdentity getClientIdentity() {
+		ClientIdentity identity = new ClientCredentials(getClientId(), getClientSecret());
```
Comment (liga-oz) [inline spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/XsuaaServiceConfigurationCustom.java:73]: I would reverse the logic -> first create ClientCertificate and check if it is valid and if not then create ClientCredentials as the X.509 becomes soon the new default auth type

**Review by liga-oz — CHANGES_REQUESTED** [review]
I really like it, well done 💯 only one thing to consider the sequence of ClientIdentity instance creation.

**Comment by nenaraab** [comment]
> I really like it, well done 💯 only one thing to consider the sequence of ClientIdentity instance creation.

Hi @liga-oz 
thanks for the hint, you're right, but in the default implementation i'm unable to change that (as getProperty is not implemented everywhere, like in spring-xsuaa) and in XsuaaServiceConfigurationCustom i kept the order from default implementation for simplicity. But we can revisit the sama as soon as ias provides credential-type as part of the binding.

**Review by liga-oz — APPROVED** [review]
I really like it, well done 💯 only one thing to consider the sequence of ClientIdentity instance creation.

**Pull request merged**